File: API/InfoTrack.Domain/Responses/SearchEnginePlotDataResponse.cs
using System.Collections.Generic;
using System.Diagnostics.CodeAnalysis;

namespace InfoTrack.Domain.Responses;

[ExcludeFromCodeCoverage]
public record SearchEnginePlotDataResponse
{
    public IEnumerable<SearchCountDataPointResponse> SearchHistory { get; set; }
    public IEnumerable<PositionDataPointResponse> PositionHistory { get; set; }
}

File: APP/src/features/search/responses/search-engine-plot-data.response.ts
import { PositionDataPointResponse } from '@/features/search/responses/position-data-point.response.ts';
import { SearchCountDataPointResponse } from '@/features/search/responses/search-count-data-point.response.ts';

export interface SearchEnginePlotDataResponse {
  searchHistory: SearchCountDataPointResponse[];
  positionHistory: PositionDataPointResponse[];
}
